python/ray/data/read_api.py

```diff
@@ -1827,6 +1827,10 @@ def convert(ds: "datasets.Dataset") -> Dataset:
         # mapping from the transformations.
         hf_ds_arrow = ds.with_format("arrow")
         ray_ds = from_arrow(hf_ds_arrow[:])
+        # To lazily apply HuggingFace feature transformations to the dataset,
+        # we apply `decode_example` to the dataset. This function only
+        # gets called when the dataset gets accessed.
+        ray_ds = ray_ds.map_batches(hf_ds_arrow.features.decode_batch)
```

Comment:
do all Huggingface Datasets have a `features` attribute? Do the features always have a `decode_batch` attribute? If so, are they always guaranteed to be a function?